### ISSUE TITLE
Control Flow Redirections?

### ISSUE THREAD

**Issue by mmstick** [redox-os/ion#566]
Should control flow keywords be able to perform and work with redirections themselves?

```sh
for x in 1...10
    echo $x
end > results.txt
```

And a theoretical `map` keyword that works similar to `for`, but reads from stdin:

```sh
cat results.txt | map x
    echo $x
end
```

**Comment by bb010g** [comment]
That last example looks really nice. I'd imagine filter just runs its body and succeeds or fails depending on the return value of the last command?

**Comment by mmstick** [comment]
Yes. That'd be the general idea of that theoretical keyword.

**Comment by huntergoldstein** [comment]
Wouldn't this require some parsing changes? From my understanding redirection only gets parsed at the pipeline level, which are contained by control structures.

In terms of wanting / not wanting, I'll note that bash has this.

**Comment by mmstick** [comment]
@huntergoldstein Yes, first being the creation of a `JobSplitter` to split a statements into jobs, before parsing those individual jobs to determine the type of job -- builtin, keyword, function, external.
